prototype2/gdgem/vmm3/BuilderVMM3.h

```diff
@@ -26,14 +27,15 @@ class BuilderVMM3 : public AbstractBuilder {
               std::shared_ptr<AbstractClusterer> x, std::shared_ptr<AbstractClusterer> y,
               uint16_t adc_threshold_x, double max_time_gap_x,
               uint16_t adc_threshold_y, double max_time_gap_y,
-              std::string dump_dir, bool dump_csv, bool dump_h5);
+              std::string dump_dir, bool dump_csv, bool dump_h5, CalibrationFile * calfile);
 
   ~BuilderVMM3() { XTRACE(INIT, DEB, "BuilderVMM2 destructor called"); }
 
   /// \todo Martin document
   ResultStats process_buffer(char *buf, size_t size) override;
 
 private:
+  CalibrationFile * calfile_;
```

Comment:
Same here. At the very least, make it default initialized to `nullptr` to avoid some nasty crashes, but you get that and more for free if you just go with a smart pointer.